### ISSUE TITLE
Deliver timer events during window drag

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/qtbase-os2#83]
Currently, timer events are not delivered to widgets when the window is being dragged with a mouse by grabbing its title bar. This breaks many things. E.g. both `qpa/qrasterwindow` and `qpa/windows` don't update their contents (because it is done by timer).

**Comment by dmik** [comment]
Timer/socket messages (WM_QT_TIMER_OR_SOCKET) are not processed because currently they are queue messages. I.e. they are processed manually by calling `WinGetMsg`. However, when one drags a window, `WinDispatchMsg` called by Qt in response to a mouse click does not return until the mouse is released. IIRC in Qt4 we used a so-called object (invisible) window to process such messages and that's why it worked — messages to windows (even to object ones) are delivered perfectly well by a nested `WinDispatchMsg` of a nested message loop that is running in PM while dragging. I will guess I will just import the relevant helper class from Qt 4.

**Comment by dmik** [comment]
The `qpa/qrasterwindow` problem is solved with `QPMObjectWindow`. However, the `qpa/windows` problem is not — as opposed to the first example, the repaint timer is not created upfront there. It is only created once the window is being moved/resized (and thus gets WM_PAINT). But since the PM message loop is inside the PM code handling WM_BUTTON1DOWN (recursively from `QEventDispatcher::processEvents`), Qt doesn't get a chance to start the timer thread — it is only started at the beginning of `QEventDispatcher::processEvents` (if there are timers).

I have to redesign the socket/timer thread machinery a bit to let this scenario work.

**Comment by dmik** [comment]
The new implementation is there and both examples work smoothly now.

It took some time to make `tst_QTimer` and `tst_QSocketNotifier` work again (those are quite severe tests). And I have an expression that they work much better with this new implementation. It also doesn't cause the select thread to be started and stopped on each `processEvents` iteration. I had to do it this way because Qt tests requres that no new timer/socket events are generated outside the `processEvents` scope. But now I solved this task in a different, better way.
